plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java

```diff
@@ -0,0 +1,336 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class ResourcesService {
+    @Autowired
+    private IResourceManager resourceManager;
+
+    @Autowired
+    private ICategoryManager categoryManager;
+
+    @Autowired
+    private IImageDimensionReader imageDimensionManager;
+
+    @Value("#{'${jacms.imageResource.allowedExtensions}'.split(',')}")
+    private List<String> imageAllowedExtensions;
+
+    @Value("#{'${jacms.attachResource.allowedExtensions}'.split(',')}")
+    private List<String> fileAllowedExtensions;
+
+    public PagedMetadata<AssetDto> listAssets(String resourceType, RestListRequest requestList) {
+        List<AssetDto> assets = new ArrayList<>();
+        try {
+            List<String> resourceIds = resourceManager.searchResourcesId(createSearchFilters(resourceType, requestList),
+                    extractCategoriesFromFilters(requestList));
+
+            for(String id : resourceIds) {
+                assets.add(convertResourceToDto(resourceType, resourceManager.loadResource(id)));
+            }
+
+        } catch (ApsException e) {
+            throw new RestServerError("Error listing image resources", e);
+        }
+
+        SearcherDaoPaginatedResult<AssetDto> paginatedResult = new SearcherDaoPaginatedResult<>(assets.size(), assets);
+        PagedMetadata<AssetDto> pagedResults = new PagedMetadata<>(requestList, paginatedResult);
+        pagedResults.setBody(requestList.getSublist(assets));
+
+        return pagedResults;
+    }
+
+    public AssetDto createAsset(String resourceType, MultipartFile file, String group, List<String> categories) {
+        BaseResourceDataBean resourceFile = new BaseResourceDataBean();
+
+        validateMimeType(resourceType, file.getContentType());
+
+        try {
+            resourceFile.setInputStream(file.getInputStream());
+            resourceFile.setFileSize(file.getBytes().length / 1000);
+            resourceFile.setFileName(file.getOriginalFilename());
+            resourceFile.setMimeType(file.getContentType());
+            resourceFile.setDescr(file.getOriginalFilename());
+            resourceFile.setMainGroup(group);
+            resourceFile.setResourceType(resourceType);
+            resourceFile.setCategories(categories.stream()
+                .map(code -> categoryManager.getCategory(code)).collect(Collectors.toList()));
+
+            ResourceInterface resource = resourceManager.addResource(resourceFile);
+            return convertResourceToDto(resourceType, resourceManager.loadResource(resource.getId()));
+        } catch (ApsSystemException e) {
+            throw new RestServerError("Error saving new resource", e);
```

Comment:
Do these exceptions get logged elsewhere? And confirm that this is the right type for the error

Reply:
These exceptions are being logged in the `RestExceptionHandler`, however are being treated as `GENERIC_ERROR`.

Ideally there should be a generic exception where you can set an HTTP status code and a message. Today, if you don't create a specific handler for each Exception, you always get a `500/GENERIC_ERROR`

Reply:
They are being logged inside the `ResourceManager` before throwing the `ApsSystemException`.